engine-tests/src/main/java/org/terasology/engine/testUtil/ModuleManagerFactory.java

```diff
@@ -2,41 +2,21 @@
 // SPDX-License-Identifier: Apache-2.0
 package org.terasology.engine.testUtil;
 
-import com.google.common.collect.Sets;
+import com.google.common.collect.ImmutableList;
 import org.terasology.engine.core.module.ModuleManager;
-import org.terasology.engine.entitySystem.stubs.StringComponent;
-import org.terasology.module.Module;
-
-import java.io.IOException;
-import java.net.URISyntaxException;
-import java.util.Collections;
-
-import static com.google.common.base.Verify.verify;
-import static org.terasology.engine.core.TerasologyConstants.ENGINE_MODULE;
+import org.terasology.gestalt.module.Module;
+import org.terasology.nui.UIWidget;
 
 public final class ModuleManagerFactory {
 
-    private ModuleManagerFactory() { }
-
-    public static ModuleManager create() throws Exception {
-        return create(false);
+    private ModuleManagerFactory() {
     }
 
-    public static ModuleManager create(boolean loadModulesFromClasspath) throws Exception {
-        ModuleManager moduleManager = new ModuleManager("", Collections.emptyList(), loadModulesFromClasspath);
-        loadUnitTestModule(moduleManager);
+    public static ModuleManager create() throws Exception {
+        // Loading screens, among other things, break when NUI classes are not added to engine.
```

Comment:
I guess this is worth its own issue. At least we should come up with a pattern or some ideas on hardening such that headless envrionments don't run into these problems...

Reply:
Yeah. And I'm on the fence about whether headless environments should have NUI. There are some "headless" tests where you don't need anything to _render,_ but maybe you're still testing to make sure the controls are there and do the right interactions.

On the other hand, on a real headless server, there's nothing trying to interact with the invisible UI, so putting it through these screens is pretty pointless.